Filter/ClassFilter.php

```diff
@@ -66,13 +66,24 @@ public function getFieldType()
      */
     public function getFieldOptions()
     {
-        return $this->getOption('choices', array(
+        if (!class_exists('Symfony\Component\Form\Extension\Core\ChoiceList\ChoiceList')) {
+            $choiceOption = array(
+                'choices' => $this->getOption('sub_classes'),
+                'choices_as_values' => true, //NEXT_MAJOR: Remove this line when drop Symfony <3.0 support
+            );
+        } else {
+            //NEXT_MAJOR: Remove this condition when drop Symfony <2.7 support
+            $choiceOption = array(
+                'choice_list' => new ChoiceList(
```

Comment:
Could you please remove `ChoiceList` altogether and do that kind of code instead: https://github.com/sonata-project/SonataAdminBundle/blob/025e8e1092ae80d305dcbeb817efb17e4ee7a4d2/Form/Type/Filter/DateTimeRangeType.php#L77

That should rid people still using sf 2.8 from some unwanted deprecation notices.